drs_downloader/cli.py

```diff
@@ -1,11 +1,124 @@
+import logging
+import uuid
+from pathlib import Path
+from typing import List
+
 import click
+import pandas as pd
+
+from drs_downloader.clients.mock import MockDrsClient
+from drs_downloader.clients.terra import TerraDrsClient
+from drs_downloader.manager import DrsAsyncManager
+
+logging.basicConfig(format='%(asctime)s %(message)s',  encoding='utf-8', level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+# these control our simulation
+NUMBER_OF_OBJECT_IDS = 10
 
 
-@click.command()
+@click.group()
 def cli():
-    """Welcome to drs_downloader."""
-    print(cli.__doc__)
+    """Copy DRS objects from the cloud to your local system ."""
+    pass
+
+
+@cli.command()
+@click.option("--silent", "-s", is_flag=True, show_default=True, default=False, help="Display nothing.")
+@click.option("--destination_dir", "-d", show_default=True, default='/tmp/testing', help="Destination directory.")
+def mock(silent: bool, destination_dir: str):
+    """Generate test files locally, without the need for server."""
+    if destination_dir:
+        destination_dir = Path(destination_dir)
+
+    # get a drs client
+    drs_client = MockDrsClient()
+
+    # assign it to a manager
+    drs_manager = DrsAsyncManager(drs_client=drs_client, show_progress=not silent)
+
+    # simulate read from manifest
+    ids_from_manifest = [str(uuid.uuid4()) for _ in range(NUMBER_OF_OBJECT_IDS)]
+
+    # get the DrsObject for the ids
+    drs_objects = drs_manager.get_objects(ids_from_manifest)
```

Comment:
does this implementation not have batching for loop where URIS are signed and passed to the downloader in batches? Is this intentional ?

Reply:
Please see manager 

https://github.com/anvilproject/drs_downloader/blob/828de8cc48947da8f212769916dad843c3042d32/drs_downloader/manager.py#L86

Reply:
Please see https://github.com/anvilproject/drs_downloader/blob/828de8cc48947da8f212769916dad843c3042d32/tests/unit/test_basic_cli.py#L18 for actual download of Terra test manifest

Reply:
so the chunker that batches the data  is just located somewhere else.